script.js

```diff
@@ -1,12 +1,121 @@
 //You can edit ALL of the code here
+//const allEpisodes = getAllEpisodes();
+
 function setup() {
-  const allEpisodes = getAllEpisodes();
-  makePageForEpisodes(allEpisodes);
+  //const allEpisodes = getAllEpisodes();
+  //makePageForEpisodes(allEpisodes);
+  //searchMovie(allEpisodes);
+  //selectAndDisplayEpisode(allEpisodes);
+  fetchAllEpisodes();
 }
-
+// Level 100
```

Comment:
Here is feedback on what is well done.

The use of class names and proper element hierarchy helps with CSS styling and structure.
This works. The search function is case-insensitive and filters episodes based on their name or summary. The fetchAllEpisodes function fetches episode data from an external API, handles the response, and calls the necessary functions to populate the page with episodes, enable search functionality, and create the select input.
The window.onload event handler ensures that the setup function is called when the window finishes loading.
The selectAndDisplayAllShows function starts defining a feature to select and display all shows, but it needs further implementation. Once you done this let me know and i review that too.

Suggestions on what can be changed

There are commented-out lines of code that should be removed to avoid confusion and clutter in the final version. The console.log statements are one example. The fetchAllEpisodes function could benefit from better error handling. Currently, it only logs errors to the console without any user-friendly feedback.
The selectAndDisplayAllShows function is incomplete and ends abruptly, leaving the implementation unfinished. It should either be completed or removed if it serves no purpose in the code.